connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufOptions.scala

```diff
@@ -46,6 +46,41 @@ private[sql] class ProtobufOptions(
   // record has more depth than the allowed value for recursive fields, it will be truncated
   // and corresponding fields are ignored (dropped).
   val recursiveFieldMaxDepth: Int = parameters.getOrElse("recursive.fields.max.depth", "-1").toInt
+
+  // Whether to deserialize empty proto3 scalar fields as their default values.
+  // When a proto3 scalar field is empty, there is ambiguity as to whether the field
+  // was never set or if the field was explicitly set to zero. By default, the library
+  // deserializes this situation as `null`, but with this flag will deserialize them
+  // as the type-specific default value.
+  //
+  // Note that this won't affect fields with the optional keyword in proto3, or
+  // any fields in proto2, as they don't have the ambiguity described above because
+  // they have presence information.
+  //
+  // As an example:
+  // ```
+  // syntax = "proto3";
+  // message Person {
+  //   string name = 1;
+  //   int64 age = 2;
+  //   optional string middle_name = 3;
+  //   optional int64 salary = 4;
+  // }
+  // ```
+  //
+  // And we have the serialized representation of the following proto:
+  // `Person(age=20, middle_name="smith")
```

Comment:
You meant 'age=0'? 
Use default value for 'middle_name' as well. That is the only case we are talking about.

Reply:
good callout, will change!

Reply:
yeahh, thanks for catching this